src/dotnet/APIView/APIViewWeb/Managers/NotificationManager.cs

```diff
@@ -87,16 +87,27 @@ public async Task NotifySubscribersOnNewRevisionAsync(ReviewListItemModel review
         /// </summary>
         /// <param name="user"></param>
         /// <param name="reviewId"></param>
+        /// <param name="state"></param> true = subscribe, false = unsubscribe
         /// <returns></returns>
-        public async Task ToggleSubscribedAsync(ClaimsPrincipal user, string reviewId)
+        public async Task ToggleSubscribedAsync(ClaimsPrincipal user, string reviewId, bool? state = null)
         {
             var review = await _reviewRepository.GetReviewAsync(reviewId);
             if (PageModelHelpers.IsUserSubscribed(user, review.Subscribers))
             {
+                if (state == true)
```

Comment:
Is UI allowing to click subscribe even if user has already subscribed?

Reply:
not UI per say. But the API can technically trigger a subscribe call when user is already subscribed.